bchrpc/server.go

```diff
@@ -1018,25 +1020,41 @@ func (s *GrpcServer) SubscribeTransactions(req *pb.SubscribeTransactionsRequest,
 				if !filter.MatchAndUpdate(txDesc.Tx, s.chainParams) {
 					continue
 				}
-				respTx := marshalTransaction(txDesc.Tx, 0, nil, 0, s.chainParams)
-				view, err := s.txMemPool.FetchInputUtxos(txDesc.Tx)
-				if err == nil {
-					for i, in := range txDesc.Tx.MsgTx().TxIn {
-						stxo := view.LookupEntry(in.PreviousOutPoint)
-						if stxo != nil {
-							respTx.Inputs[i].Value = stxo.Amount()
-							respTx.Inputs[i].PreviousScript = stxo.PkScript()
-
-							_, addrs, _, err := txscript.ExtractPkScriptAddrs(stxo.PkScript(), s.chainParams)
-							if err == nil && len(addrs) > 0 {
-								respTx.Inputs[i].Address = addrs[0].String()
+
+				toSend := &pb.TransactionNotification{}
+
+				// Serialize raw tx using Bitcoin protocol encoding
+				if serializeTx {
+					tx := txDesc.Tx
```

Comment:
NIT: would make this a method, that makes the flow a lot easier to read! Basically each if/else should just call the corresponding method and return type/error.